<launch>
    <include file="$(find xycar_motor)/launch/xycar_motor_a2.launch" />
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
        <param name="video_device" value="/dev/video0" />
        <param name="autoexposure" value="false"/>
        <param name="exposure" value="48"/>
        <param name="image_width" value="640" />
        <param name="image_height" value="480" />
        <param name="pixel_format" value="yuyv" />
        <!--param name="camera_frame_id" value="usb_cam" /-->
	<param name="camera_frame_id" value="map" />
	<param name="camera_name" value="usb_cam" />
        <param name="io_method" value="mmap"/>
	<param name="camera_info_url" value="file://$(find final_project)/calibration/usb_cam.yaml" />
    </node>
    <node name="xycar_lidar" pkg="xycar_lidar" type="xycar_lidar" output="screen">
	<param name = "serial_port" type="string" value="/dev/ttyRPL"/>
	<param name = "serial_baudrate" type="int" value="115200"/>
	<param name = "frame_id" type="string" value="laser"/>
	<param name = "inverted" type="bool" value="false"/>
	<param name = "angle_compensate" type="bool" value="true"/>
    </node>

    <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
    	<param name="marker_size" type="double" value="9.0" />
    	<param name="max_new_marker_error" type="double" value="0.05" />
	<param name="max_track_error" type="double" value="0.05" />
	<param name="output_frame" type="string" value="map" />

	<remap from="camera_image" to="/usb_cam/image_raw" />
	<remap from="camera_info" to="/usb_cam/camera_info" />
    </node>

    <node name="auto_driver" pkg="final_project" type="final_project.py" output="screen"/>
</launch>
